src/drivers/phy.c

```diff
@@ -71,9 +65,10 @@ void mg_phy_init(struct mg_phy *phy, uint8_t phy_addr, uint8_t config) {
       phy->write_reg(phy_addr, MG_PHY_DP83x_REG_RCSR, MG_BIT(7) | MG_BIT(0));
     } else if (id1 == MG_PHY_KSZ8x) {
       // Disable isolation (override hw, it doesn't make sense at this point)
-      phy->write_reg(  // #2848, some NXP boards set ISO, even though
-          phy_addr, MG_PHY_REG_BCR,  // docs say they don't
-          phy->read_reg(phy_addr, MG_PHY_REG_BCR) & (uint16_t) ~MG_BIT(10));
+      // #2848, some NXP boards set ISO, even though docs say they don't
+      uint16_t bcr = phy->read_reg(phy_addr, MG_PHY_REG_BCR);
+      bcr &= ~(((uint16_t) 1U) << 10U);
+      phy->write_reg(phy_addr, MG_PHY_REG_BCR, bcr);
```

Comment:
This change is breaking tests.
What is the reason for it ? Looks like it is just style. Inverting bits of a uint16 makes it an unsigned int, that is why it was casted that way, after silencing all compilers at that time, when doing that change.
Why 3 sentences instead of a clear 1 is better ? I think the read-modify-right operation is clear with one sentence; anyway, at least keep the way the bit is cleared, I do think it is easier to read, and it definitely does not trigger test failures

It is not related to the RNG change, anyway

Reply:
The reason is MSVC 98 (our C89 compiler) giving warning about the current code about conversion size mismatch. Yeah, not related to RNG

Reply:
Jeeze... please see chat